File: app/src/main/java/com/duy/interpreter/exceptions/ParsingException.java
package com.duy.interpreter.exceptions;


import com.duy.interpreter.linenumber.LineInfo;

public class ParsingException extends Exception {
    private static final long serialVersionUID = -5592261827134726938L;
    public LineInfo line;

    public ParsingException(LineInfo line, String message) {
        super(message);
        this.line = line;
    }

    public ParsingException(LineInfo line) {
        super();
        this.line = line;
    }

    @Override
    public String toString() {
        return line + ":" + getMessage();
    }
}

File: app/src/main/java/com/duy/pascal/interperter/parse_exception/ParsingException.kt
package com.duy.pascal.interperter.parse_exception


import com.duy.pascal.interperter.linenumber.LineInfo

/**
 * Exception when parse syntax of the program
 */
open class ParsingException : Exception {

    /**
     * The lineInfo of lineInfo error
     */
    var lineInfo: LineInfo? = null

    constructor(lineInfo: LineInfo?, message: String) : super(message) {
        this.lineInfo = lineInfo
    }

    constructor(lineInfo: LineInfo?) : super() {
        this.lineInfo = lineInfo
    }

    override fun toString(): String {
        return lineInfo.toString() + ":" + message
    }

    open val isAutoFix: Boolean
        get() = false
}
